pinot-controller/src/main/java/org/apache/pinot/controller/ControllerLeadershipManager.java

```diff
@@ -18,50 +18,66 @@
  */
 package org.apache.pinot.controller;
 
+import java.util.HashMap;
 import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.concurrent.ThreadSafe;
 import org.apache.helix.HelixManager;
-import org.apache.helix.api.listeners.ControllerChangeListener;
 import org.apache.pinot.common.metrics.ControllerGauge;
 import org.apache.pinot.common.metrics.ControllerMetrics;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 /**
- * Single place for listening on controller changes
- * This should be created at controller startup and everyone who wants to listen to controller changes should subscribe
+ * Single place for listening on controller changes.
+ * This should be created at controller startup and everyone who wants to listen to controller changes should subscribe.
  */
+@ThreadSafe
 public class ControllerLeadershipManager {
-
   private static final Logger LOGGER = LoggerFactory.getLogger(ControllerLeadershipManager.class);
 
-  private HelixManager _helixManager;
-  private ControllerMetrics _controllerMetrics;
-  private volatile boolean _amILeader = false;
+  private final HelixManager _helixManager;
+  private final ControllerMetrics _controllerMetrics;
 
-  private Map<String, LeadershipChangeSubscriber> _subscribers = new ConcurrentHashMap<>();
+  private Map<String, LeadershipChangeSubscriber> _subscribers = new HashMap<>();
+  private boolean _amILeader = false;
```

Comment:
It might be good to mark it as `volatile`?

Reply:
It does not need to be as the methods are synchronized